gtsam/geometry/FundamentalMatrix.h

```diff
@@ -0,0 +1,183 @@
+ * left rotation U, a scalar s, and a right rotation V.
+ */
+class GTSAM_EXPORT FundamentalMatrix {
+ private:
+  Rot3 U_;    ///< Left rotation
+  double s_;  ///< Scalar parameter for S
+  Rot3 V_;    ///< Right rotation
+
+ public:
+  /// Default constructor
+  FundamentalMatrix() : U_(Rot3()), s_(1.0), V_(Rot3()) {}
+
+  /**
+   * @brief Construct from U, V, and scalar s
+   *
+   * Initializes the FundamentalMatrix with the given left rotation U,
+   * scalar s, and right rotation V.
+   *
+   * @param U Left rotation matrix
+   * @param s Scalar parameter for the fundamental matrix
+   * @param V Right rotation matrix
+   */
+  FundamentalMatrix(const Rot3& U, double s, const Rot3& V)
+      : U_(U), s_(s), V_(V) {}
+
+  /**
+   * @brief Construct from a 3x3 matrix using SVD
+   *
+   * Initializes the FundamentalMatrix by performing SVD on the given
+   * matrix and ensuring U and V are not reflections.
+   *
+   * @param F A 3x3 matrix representing the fundamental matrix
+   */
+  FundamentalMatrix(const Matrix3& F);
+
+  /// Return the fundamental matrix representation
+  Matrix3 matrix() const;
+
+  /// @name Testable
+  /// @{
+  /// Print the FundamentalMatrix
+  void print(const std::string& s = "") const;
+
+  /// Check if the FundamentalMatrix is equal to another within a
+  /// tolerance
+  bool equals(const FundamentalMatrix& other, double tol = 1e-9) const;
+  /// @}
+
+  /// @name Manifold
+  /// @{
+  enum { dimension = 7 };  // 3 for U, 1 for s, 3 for V
+  inline static size_t Dim() { return dimension; }
+  inline size_t dim() const { return dimension; }
+
+  /// Return local coordinates with respect to another FundamentalMatrix
+  Vector localCoordinates(const FundamentalMatrix& F) const;
+
+  /// Retract the given vector to get a new FundamentalMatrix
+  FundamentalMatrix retract(const Vector& delta) const;
+  /// @}
+};
+
+/**
+ * @class SimpleFundamentalMatrix
+ * @brief Class for representing a simple fundamental matrix.
+ *
+ * This class represents a simple fundamental matrix, which is a
+ * parameterization of the essential matrix and focal lengths for left and right
+ * cameras. Principal points are not part of the manifold but a convenience.
+ */
+class GTSAM_EXPORT SimpleFundamentalMatrix {
+ private:
+  EssentialMatrix E_;  ///< Essential matrix
+  double fa_;          ///< Focal length for left camera
+  double fb_;          ///< Focal length for right camera
+  Point2 ca_;          ///< Principal point for left camera
+  Point2 cb_;          ///< Principal point for right camera
+
+ public:
+  /// Default constructor
+  SimpleFundamentalMatrix()
+      : E_(), fa_(1.0), fb_(1.0), ca_(0.0, 0.0), cb_(0.0, 0.0) {}
+
+  /// Construct from essential matrix and focal lengths
+  SimpleFundamentalMatrix(const EssentialMatrix& E,  //
+                          double fa, double fb,
+                          const Point2& ca = Point2(0.0, 0.0),
+                          const Point2& cb = Point2(0.0, 0.0))
```

Comment:
so these are not in pixel space, but with respect to image center?

Reply:
ca and cb are the pixel centers. If you do not provide ca and cb, then your are promising to provide coordinates with respect to the pixel center, yes.

Reply:
I think it would be better to have ca and cb be required inputs so users don't unknowingly make this promise

Reply:
OK, will do